### PR TITLE
Refactoring/renaming/best practices 

### PR DESCRIPTION
**Pull request by apurvis**
here's a bunch of refactor junk i kept out of the main PR over on https://github.com/lumoslabs/broadside/pull/16 

the most important thing for the benefit of next person reading the code is probably the variable renames to match what AWS objects are actually called, though there's some general best practices improvements as well and some hounding that was bothering me but that i didn't want cluttering up the main PR

note this PR is to be merged into https://github.com/lumoslabs/broadside/pull/16 or master after we merge that one though i'm not in a rush; i could wait for @matl33t to write some better specs

@matl33t @rfroetscher

### CODE REVIEW

File: lib/broadside/deploy/ecs_deploy.rb [hunk 1]
```diff
@@ -202,85 +202,78 @@ def create_task_definition(name, options = {})
...
+        seen_event = nil
+
+        w.before_wait do |attempt, response|
+          debug "(#{attempt}/#{w.max_attempts}) Polling ECS for events..."
+          # skip first event since it doesn't apply to current request
+          if response.services[0].events.first && response.services[0].events.first.id != seen_event && attempt > 1
+            seen_event = response.services[0].events.first.id
+            debug(response.services[0].events.first.message)
           end
-        rescue Aws::Waiters::Errors::TooManyAttemptsError
```
Comment (matl33t) [inline lib/broadside/deploy/ecs_deploy.rb]: what's the reason for removing this?

File: spec/broadside/deploy/ecs_deploy_spec.rb [hunk 2]
```diff
@@ -62,8 +62,8 @@
     context 'with an existing service' do
       let :existing_service do
         {
-          service_arn: "arn:aws:ecs:us-east-1:447374670232:service/events_test_ecs_script_2",
-          service_name: "events_test_ecs_script_2",
+          service_arn: "arn:aws:ecs:us-east-1:447374670232:service/#{task_name}",
```
Comment (matl33t) [inline spec/broadside/deploy/ecs_deploy_spec.rb]: where is task_name coming from?

File: lib/broadside/deploy/ecs_deploy.rb (hunk 1 above)
Comment (apurvis) [inline lib/broadside/deploy/ecs_deploy.rb]: i was thinking the exception would bubble up and create an exception anyways, just of the type `Aws::Waiters::Errors::TooManyAttemptsError` - which is actually more accurate - than the `Broadside::Error` or whatever.

in general i'm not a fan of squashing a bunch of useful information about exceptions by removing their class and stuffing them in a single exception class; it's caused me some pain debugging.

File: spec/broadside/deploy/ecs_deploy_spec.rb (hunk 2 above)
Comment (apurvis) [inline spec/broadside/deploy/ecs_deploy_spec.rb]: line 71.  (rspec `let` can work out of order)

**Comment by apurvis** [comment]
i refactored all the ECS management to it's own class on https://github.com/lumoslabs/broadside/pull/20

now the `EcsDeploy` class just handles higher level stuff

File: lib/broadside/deploy/ecs_manager.rb [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      def service_exists?(cluster, family)
+        services = ecs.describe_services(cluster: cluster, services: [family])
+        services.failures.empty? && !services.services.empty?
+      end
+
+      private
+
+      def all_results(method, key, args = {})
+        page = ecs.public_send(method, args)
+        results = page.send(key)
```
Comment (rfroetscher) [inline lib/broadside/deploy/ecs_manager.rb]: `public_send`

Comment (apurvis) [inline lib/broadside/deploy/ecs_manager.rb]: ha this time i thought i already did that but maybe i didn't push or something...

**Comment by apurvis** [comment]
i tested all this stuff and it seems good to go so i'm going to close this inter branch PR and open a new clean one onto master.

**Pull request closed without merging**